torch_geometric/nn/models/node2vec.py

```diff
@@ -8,7 +8,59 @@
 from torch_geometric.typing import WITH_PYG_LIB, WITH_TORCH_CLUSTER
 from torch_geometric.utils import sort_edge_index
 from torch_geometric.utils.num_nodes import maybe_num_nodes
-from torch_geometric.utils.sparse import index2ptr
+from torch_geometric.utils.sparse import index2ptr, ptr2index
+
+####################################################################################### { define a weighted random walk that care the data.attr
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+def random_walk_weighted(rowptr, col, weight, batch, walk_length, p, q):
```

Comment:
@rusty1s do we have support for weighted walk in `pyg-lib`? For some reason I rmember seeing it in some roadmap.

Reply:
@wsad1 @rusty1s 
Thank you for your feedback. 
I have incorporated your original random walk C++ code into the weighted random walk by computing the probabilities of existing edges using their weights in each batch. 
By default, the **_fastmode_** is set to True, which means that the weighted walk has the same time complexity as the random walk. 

However, if the user sets **_fastmode_** to False, the following weighted random walk in Python is implemented. 
`def random_walk_weighted(rowptr, col, weight, batch, walk_length, p, q)`
This allows the user to customize the code of the weighted walk, but it results in slower execution time.

` model = Node2Vec(
        data.edge_index,
        embedding_dim=128,
        walk_length=20,
        context_size=10,
        edge_attr=data.edge_attr,  # Add data attribute to the input
        fast_mode=True,  # Fast Estimate or Exact Computation
        walks_per_node=10,
        num_negative_samples=1,
        p=1,
        q=1,
        sparse=True,
    ).to(device)`